### PR TITLE
Add jdk17 to the build pipelines

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Also finish removing jdk15

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

File: buildenv/jenkins/variables/defaults.yml [hunk 1]
```diff
@@ -167,7 +172,8 @@ boot_jdk_default:
       8: '8'
       11: '11'
       16: '16'
-      next: '16'
+      17: '16'
+      next: '17'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml:176]: next needs to stay at 16 until 17 is released. It won't require 17 until after release.

Comment (AdamBrousseau) [reply buildenv/jenkins/variables/defaults.yml]: Updated. Just trying to push the repo up in order to try a build.

**Comment by pshipton** [comment]
jenkins build zlinux jdk16

**Comment by pshipton** [comment]
jenkins compile zlinux jdk16

**Review by pshipton — APPROVED** [review]

**Comment by pshipton** [comment]
jenkins compile zlinux jdk17

File: buildenv/jenkins/variables/defaults.yml (hunk 1 above)
Comment (pshipton) [reply buildenv/jenkins/variables/defaults.yml]: Do you want me to keep this un-merged until the repo is ready and we can do a jdk17 PR?

**Comment by AdamBrousseau** [comment]
jenkins compile zlinux jdk17

**Comment by AdamBrousseau** [comment]
jenkins compile zlinux jdk17 depends ibmruntimes/openj9-openjdk-jdk17#openj9-staging

**Comment by AdamBrousseau** [comment]
jenkins compile zlinux jdk17 depends ibmruntimes/openj9-openjdk-jdk17#openj9-staging

**Pull request merged**